app/content/drill/[id]/leaderboard.js

```diff
@@ -1,27 +1,58 @@
 import { Link, useLocalSearchParams, usePathname } from "expo-router";
-import { useEffect, useState } from "react";
+import { useState } from "react";
 import { ScrollView, View } from "react-native";
 import { Avatar, Icon, List, Text } from "react-native-paper";
 import { numTrunc } from "~/Utility";
 import EmptyScreen from "~/components/emptyScreen";
 import ErrorComponent from "~/components/errorComponent";
 import Loading from "~/components/loading";
 import RefreshInvalidate from "~/components/refreshInvalidate";
-import { currentAuthContext } from "~/context/Auth";
-import { updateLeaderboard } from "~/hooks/updateLeaderboard";
 import { useAttempts } from "~/hooks/useAttempts";
 import { useDrillInfo } from "~/hooks/useDrillInfo";
 import { useLeaderboard } from "~/hooks/useLeaderboard";
 import { useUserInfo } from "~/hooks/useUserInfo";
 
+function getLeaderboardRanks(
+  orderedLeaderboard,
+  leaderboardAttempts,
+  mainOutputAttempt,
+) {
+  let leaderboardRanks = [];
+  let currentRank = 1;
+
+  for (let i = 0; i < orderedLeaderboard.length; i++) {
+    const userId = orderedLeaderboard[i];
+
+    const attemptValue = numTrunc(
+      leaderboardAttempts[userId][mainOutputAttempt].value,
+    );
+    //base case for a clear number 1
+    if (i == 0) {
+      leaderboardRanks.push(currentRank);
+    }
+
+    //case for a tie
+    else if (attemptValue == prevAttemptValue) {
+      leaderboardRanks.push(currentRank);
```

Comment:
Stylistic suggestion, but instead of displaying the same number for people with the same ranks, why not just leave it blank?
1. Player A
    Player B
2. Player C

Reply:
I prefer having the rank for everyone, makes it more of a leaderboard and if there is a big tie somewhere I think it might look strange. Like it could go off the screen. additionally, me and Jeff based the rank display off a speed run leaderboard that had this approach. 

1. Player A
    Player B
    Player C
    Player D
    Player E
    Player F
2. Player G

![image](https://github.com/Golf-Drill-Challenge-App/Golf-App/assets/91503842/3fd97c7a-b0ed-486d-8c39-84a195f1548a)

Reply:
alr